### PR TITLE
Update toolmanager_sgskip.py

### PR DESCRIPTION
**Pull request by valtron**
## PR Summary

This example seems only works with backends that have a toolbar (`fig.canvas.manager.toolbar is not None`). Because different platforms may have different default backends, it may not be obvious to a user why a script using these features fails on another machine. Explicitly setting the backend at the top ensures a backend that has a toolbar is used.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by QuLogic** [comment]
That is not true; the toolmanager interface is implemented in Tk, GTK, and Qt. Defaulting to Tk may be the best option though, as that should be available everywhere.

PS, please give your commits/PRs descriptive messages.

**Comment by jklymak** [comment]
@valtron, can you explain where this did _not_ work?  Maybe what would be better is a comment at the top of the example that explains the type of backend required for the toolmanager?

**Comment by valtron** [comment]
@QuLogic @jklymak Sorry for the lack of details, I didn't have time until today to track down what was happening, because it was failing on someone else's machine.

A script that was using the toolmanager/toolbar was failing on a Mac because on there, `matplotlib.get_backend() == 'MacOSX'`, and that one didn't have a toolbar.

File: examples/user_interfaces/toolmanager_sgskip.py [hunk 1]
```diff
@@ -13,6 +13,10 @@
 Using `matplotlib.backend_managers.ToolManager`
 """
 
+import matplotlib
+# Not all backends have a toolbar
```
Comment (QuLogic) [inline examples/user_interfaces/toolmanager_sgskip.py:17]: They all have a toolbar; not all have toolmanager.
```suggestion
# Not all backends implement toolmanager.
```

Comment (valtron) [reply examples/user_interfaces/toolmanager_sgskip.py:17]: Sorry, I'm not familiar with the matplotlib terminology for these things. But what I saw was that on the `MacOSX` backend, `canvas.manager.toolmanager` actually did exist (was not `None`), and it was `canvas.manager.toolbar` that was `None`. For another example, on the `Agg` backend, both are `None` :/

Comment (QuLogic) [reply examples/user_interfaces/toolmanager_sgskip.py:17]: This example uses ToolManager, so `canvas.manager.toolbar` will be `None` on all backends. The `Agg` backend does not show anything, so has no toolbars of any kind.

Comment (valtron) [reply examples/user_interfaces/toolmanager_sgskip.py:17]: If `canvas.manager.toolbar` is `None`, then [this line](https://github.com/matplotlib/matplotlib/blob/3f823871f123a3d3f886c347f80addd7c21a9d2f/examples/user_interfaces/toolmanager_sgskip.py#L87) will throw an exception so the example shouldn't work at all?

Comment (QuLogic) [reply examples/user_interfaces/toolmanager_sgskip.py:17]: Ah sorry, I misremembered. `.toolbar` will be set to either a toolbar2 or a toolmanager, but it's _not_ the toolbar that's really missing.

Comment (valtron) [reply examples/user_interfaces/toolmanager_sgskip.py:17]: Right, so the issue is that on the `MacOSX` backend, `canvas.manager.toolbar` _was_ `None` despite a graph being shown (unlike `Agg`, for which the behaviour makes sense and I would expect the failure).

Maybe I could change the comment to something less specific, like "Tools are only available on certain backends"?

**Pull request closed without merging**